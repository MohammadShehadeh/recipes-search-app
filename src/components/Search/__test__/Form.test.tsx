import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import { Form } from '..';

const submitHandlerMock = jest.fn();

describe('Form', () => {
  it('should render to match snapshot', () => {
    const tree = renderer.create(<Form submitHandler={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle submit button correctly', () => {
    render(<Form submitHandler={submitHandlerMock} />);

    const buttonElement = screen.queryAllByRole('button');
    const inputElement = screen.getByPlaceholderText('Type recipe name...');

    userEvent.type(inputElement, 'value');
    userEvent.click(buttonElement[0]);
    expect(submitHandlerMock).toHaveBeenCalledWith('value');
  });

  it('should handle clear button correctly', () => {
    render(<Form submitHandler={submitHandlerMock} />);

    const buttonElement = screen.queryAllByRole('button');
    const inputElement = screen.getByPlaceholderText('Type recipe name...');

    userEvent.type(inputElement, 'value');
    userEvent.click(buttonElement[1]);
    expect(inputElement).toHaveValue('');
    expect(submitHandlerMock).toHaveBeenCalledWith('');
  });
});
