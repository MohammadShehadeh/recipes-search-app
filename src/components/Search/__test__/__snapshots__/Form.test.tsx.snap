// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render to match snapshot 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <fieldset
    className="fieldset"
  >
    <legend
      className="legend"
    >
      Search For Recipes:
    </legend>
    <div
      className="group"
    >
      <input
        className="input"
        id="search"
        name="Search"
        placeholder="Type recipe name..."
        type="search"
      />
      <button
        className="submit"
        title="Search For Recipes"
        type="submit"
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.5 14C7 14 5 12 5 9.5S7 5 9.5 5 14 7 14 9.5 12 14 9.5 14zm6 0h-.8l-.3-.3c1-1.1 1.6-2.6 1.6-4.2C16 5.9 13.1 3 9.5 3S3 5.9 3 9.5 5.9 16 9.5 16c1.6 0 3.1-.6 4.2-1.6l.3.3v.8l5 5 1.5-1.5-5-5z"
          />
        </svg>
      </button>
    </div>
    <button
      className="clear"
      onClick={[Function]}
      type="reset"
    >
      Clear
    </button>
  </fieldset>
</form>
`;
